src/libdredd/src/mutation_replace_binary_operator.cc

```diff
@@ -50,6 +50,19 @@ MutationReplaceBinaryOperator::MutationReplaceBinaryOperator(
     const clang::BinaryOperator& binary_operator)
     : binary_operator_(binary_operator) {}
 
+bool MutationReplaceBinaryOperator::IsValidReplacementOperator(
+    clang::BinaryOperatorKind op) const {
+  const auto* lhs_type =
+      binary_operator_.getLHS()->getType()->getAs<clang::BuiltinType>();
+  const auto* rhs_type =
+      binary_operator_.getRHS()->getType()->getAs<clang::BuiltinType>();
+  return !((lhs_type->isFloatingPoint() || rhs_type->isFloatingPoint()) &&
```

Comment:
I would:

- add assertions that both lhs_type and rhs_type are either integer or floating-point (this should be true by construction, but it's good to confirm); use the && trick to have a suitable string associated with the assertions.
- add a comment noting that certain binary operators are not compatible with floating-point arguments.

Reply:
Done.